PHPCompatibility/Tests/Syntax/NewInterpolatedStringDereferencingUnitTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\Syntax;
+
+use PHPCompatibility\Tests\BaseSniffTest;
+
+/**
+ * Test the InterpolatedStringDereferencing sniff.
+ *
+ * @group interpolatedStringDereferencing
+ * @group syntax
+ *
+ * @covers \PHPCompatibility\Sniffs\Syntax\NewInterpolatedStringDereferencingSniff
+ *
+ * @since 10.0.0
+ */
+class NewInterpolatedStringDereferencingUnitTest extends BaseSniffTest
+{
+
+    /**
+     * Test that an error is thrown for interpolated string dereferencing prior to PHP 8.
+     *
+     * @dataProvider dataInterpolatedStringDereferencing
+     *
+     * @param int $line Line number where the error should occur.
+     *
+     * @return void
+     */
+    public function testInterpolatedStringDereferencing($line)
+    {
+        $file = $this->sniffFile(__FILE__, '7.4');
+        $this->assertError($file, $line, 'Dereferencing of interpolated strings is not supported in PHP 7.4 or earlier');
+    }
+
+    /**
+     * Data provider.
+     *
+     * @see testInterpolatedStringDereferencing
+     *
+     * @return array
+     */
+    public function dataInterpolatedStringDereferencing()
+    {
+        return [
+            [24],
+            [25],
+            [26],
+        ];
+    }
+
+    /**
+     * Verify the sniff does not throw false positives for valid code.
+     *
+     * @dataProvider dataNoFalsePositives
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testNoFalsePositives($line)
+    {
+        $file = $this->sniffFile(__FILE__, '7.4');
+        $this->assertNoViolation($file, $line);
+    }
+
+    /**
+     * Data provider.
+     *
+     * @see testNoFalsePositives()
+     *
+     * @return array
+     */
+    public function dataNoFalsePositives()
+    {
+        return [
+
```

Comment:
Superfluous blank lines at the start and end of the array.

Also, as the trailing comments don't just apply to just that one line, but to all lines in each block, they should probably be moved to be above each block.

```php
            // No interpolation, no dereferencing.
            [4],
            [5],
            [6],
```